src/libraries/System.Drawing.Common/src/System/Drawing/ImageInfo.cs

```diff
@@ -140,22 +132,61 @@ public int FrameCount
             }
 
             /// <summary>
-            /// The delay associated with the frame at the specified index.
+            /// The total animation time of the image, in milliseconds.
             /// </summary>
-            public int FrameDelay(int frame)
+            private int TotalAnimationTime
             {
-                return _frameDelay![frame];
+                get
+                {
+                    if (Animated)
+                    {
+                        return _frameEndTimes[_frameCount - 1];
+                    }
+
+                    return 0;
+                }
             }
 
-            internal int FrameTimer
+            /// <summary>
+            /// Advance the animation by the specified number of milliseconds. If the advancement
+            /// progresses beyond the end time of the current Frame, then <see cref="Frame"/> will
+            /// be updated and the <see cref="FrameChangedHandler"/> will be called. Subscribed
+            /// handlers often use that event to call <see cref="ImageAnimator.UpdateFrames(Image)"/>.
+            /// <para>
+            /// If the animation progresses beyond the end of the image's total animation time,
+            /// the animation will loop.
+            /// </para>
+            /// </summary>
+            /// <remarks>
+            /// This animation does not respect a GIF's specified number of animation repeats;
+            /// instead, animations loop indefinitely.
+            /// </remarks>
+            /// <param name="milliseconds">The number of milliseconds to advance the animation by</param>
+            public void AdvanceAnimationBy(int milliseconds)
             {
-                get
+                int oldFrame = _frame;
+                _frameTimer += milliseconds;
+
+                if (_frameTimer > TotalAnimationTime)
                 {
-                    return _frameTimer;
+                    _frameTimer %= TotalAnimationTime;
```

Comment:
>_frameTimer %= TotalAnimationTime [](http://example.com/codeflow?start=20&length=33)

Is it possible AdvanceAnimationBy can be called while Animated is false? I am asking as we could be dividing by zero here.

Reply:
It was guarded in `ImageAnimator`, but guarding here would be an improvement.